selftests/unit/test_utils_network.py

```diff
@@ -33,5 +34,39 @@ def test_release_port(self):
         self.assertNotIn(22, tracker.retained_ports)
 
 
+class FreePort(unittest.TestCase):
+
+    def test_is_port_free(self):
+        port = network.find_free_port()
+        self.assertTrue(network.is_port_free(port, "localhost"))
```

Comment:
I don't claim to have a perfect solution for this, but this (as is `network.find_free_port()`) is susceptible to race conditions if other software is running on the same system.

One possibility, though, that just crossed my mind, is the fact that we can use any address on 127.0.0.1/8 range, and something like that can be also true for IPv6.  In theory, if we pick an address other than 127.0.0.1, we at least, minimize a lot the possibility of clashes and thus race conditions.